File: 1692/h/h.py
from sys import stdin
from collections import defaultdict, Counter
def input(): return stdin.readline().strip()


def read_int():
    return int(input())


def read_ints():
    return map(int, input().split())


t = read_int()
for case_num in range(t):
    n = read_int()
    x = list(read_ints())
    l = 1
    r = 1
    ans = 1
    choice = x[0]

    # Solution 1: solve for each number separately
    # p = defaultdict(list)
    # for i, xi in enumerate(x):
    #     p[xi].append(i)

    # for num in p:
    #     mini = 0
    #     minv = 1000000
    #     for i, vi in enumerate(p[num]):
    #         now = 2 * i - vi
    #         if i > 0:
    #             cur = now - minv + 1
    #             if cur > ans:
    #                 ans = cur
    #                 choice = num
    #                 l = p[num][mini] + 1
    #                 r = vi + 1
    #         if now < minv:
    #             minv = now
    #             mini = i

    # Solution 2: solve all numbers together
    c = Counter()
    d = defaultdict(lambda: 1000000)
    e = defaultdict(int)

    ans = 1
    l = 1
    r = 1
    choice = x[0]

    for i, num in enumerate(x):
        c[num] += 1
        diff = c[num] - (i + 1 - c[num])
        if diff - d[num] + 1 > ans:
            ans = diff - d[num] + 1
            l = e[num] + 1
            r = i + 1
            choice = x[i]
        if diff < d[num]:
            d[num] = diff
            e[num] = i

    print(choice, l, r)

File: 1692/h/h.cc
#include <cstdio>
#include <iostream>
#include <map>
#include <vector>

using namespace std;
template <typename T>
void read(T &x) {
  x = 0;
  char c = getchar();
  T sig = 1;
  for (; !isdigit(c); c = getchar())
    if (c == '-') sig = -1;
  for (; isdigit(c); c = getchar()) x = (x << 3) + (x << 1) + c - '0';
  x *= sig;
}

class Solution {
 public:
  void solve() {
    int n;
    read(n);

    int l = 1, r = 1, ans = 1, choice = 0;
    map<int, vector<int>> p;
    for (int i = 0; i < n; ++i) {
      int x;
      read(x);
      if (!choice) choice = x;
      p[x].push_back(i);
    }

    for (auto [num, v] : p) {
      int mini = 0, minv = 1000000;
      for (int i = 0; i < v.size(); ++i) {
        int now = 2 * i - v[i];
        if (i > 0) {
          int cur = now - minv + 1;
          if (cur > ans) {
            ans = cur;
            choice = num;
            l = v[mini] + 1;
            r = v[i] + 1;
          }
        }
        if (now < minv) {
          minv = now;
          mini = i;
        }
      }
    }

    printf("%d %d %d\n", choice, l, r);
  }
};

int main() {
  ios::sync_with_stdio(false);
  cin.tie(0);
  int t;
  read(t);
  while (t--) {
    Solution solution = Solution();
    solution.solve();
  }
}